crates/uv/tests/it/tool_install.rs

```diff
@@ -1817,6 +1817,270 @@ fn tool_install_unnamed_package() {
     "###);
 }
 
+/// Test installing a tool with a Git requirement.
+#[test]
+#[cfg(feature = "git")]
+fn tool_install_git() {
+    let context = TestContext::new("3.12").with_filtered_exe_suffix();
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    // Avoid removing `git` from PATH
+    let git_bin_path = which::which("git").expect("Failed to find `git` executable.");
+    let git_path = git_bin_path
+        .parent()
+        .expect("Failed to find `git` executable directory.");
+    let path = std::env::join_paths(&[bin_dir.to_path_buf(), git_path.to_path_buf()]).unwrap();
+
+    // Unnamed Git Install
+    uv_snapshot!(context.filters(), context.tool_install()
+        .arg("git+https://github.com/psf/black@24.2.0")
+        .env(EnvVars::UV_TOOL_DIR, tool_dir.as_os_str())
+        .env(EnvVars::XDG_BIN_HOME, bin_dir.as_os_str())
+        .env(EnvVars::PATH, path.as_os_str()), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 6 packages in [TIME]
+    Prepared 4 packages in [TIME]
+    Installed 6 packages in [TIME]
+     + black==24.2.0 (from git+https://github.com/psf/black@6fdf8a4af28071ed1d079c01122b34c5d587207a)
```

Comment:
We should use something smaller here, black is convenient but Git plus the black dependencies is noticeable in test suite performance.